crates/workspace/src/pane.rs

```diff
@@ -2924,12 +2924,9 @@ impl Pane {
                             let moved_right = ix > from_ix;
                             let ix = if moved_right { ix - 1 } else { ix };
                             let is_pinned_in_to_pane = this.is_tab_pinned(ix);
-                            let is_at_same_position = ix == from_ix;
 
-                            if is_at_same_position
-                                || (moved_right && is_pinned_in_to_pane)
-                                || (!moved_right && !is_pinned_in_to_pane)
-                                || (!moved_right && was_pinned_in_from_pane)
+                            if (was_pinned_in_from_pane && is_pinned_in_to_pane)
+                                || (!was_pinned_in_from_pane && !is_pinned_in_to_pane)
```

Comment:
I generalized the logic when moving tabs within the same pane. This feels more sound.